ruby/lib/minitest/queue/test_time_reporter.rb

```diff
@@ -0,0 +1,69 @@
+require 'minitest/reporters'
+
+module Minitest
+  module Queue
+    class TestTimeReporter < Minitest::Reporters::BaseReporter
+      include ::CI::Queue::OutputHelpers
+
+      def initialize(build:, limit: nil, percentile: nil, **options)
+        super(options)
+        @test_time_hash = build.fetch
+        @limit = limit
+        @percentile = percentile
+        @success = true
+      end
+
+      def report
+        return if limit.nil? || test_time_hash.empty?
+
+        puts '+++ Test Time Report'
+
+        if offending_tests.empty?
+          msg = "The #{humanized_percentile} of test execution time is within #{limit} milliseconds."
+          puts green(msg)
+          return
+        end
+
+        @success = false
+        puts <<~EOS
+          #{red("Detected #{offending_tests.length} test(s) over the desired time limit.")}
```

Comment:
Nitpick: I know that we don't have Rubocop configured for this project, but using `.size` on arrays is usually preferred over `.length` (with `.length` used for string lengths).